src/index.js

```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import ReactDOM from "react-dom/client";
+import "./index.css";
+import Home from "./pages/Home";
+import Video from "./pages/Video";
+import { RouterProvider, createBrowserRouter } from "react-router-dom";
+
+const router = createBrowserRouter([
+  {
+    path: "/",
+    element: <Home />,
+  },
+  {
+    path: "/video/:id",
+    element: <Video />,
+  },
+]);
+
+const root = ReactDOM.createRoot(document.getElementById("root"));
+root.render(
+  <React.StrictMode>
+    <RouterProvider router={router} />
```

Comment:
with other React apps, I'm used to seeing <App/> so now wat is the RouterProvider

Reply:
As of React Router v6, they now use this context provider to registry routes. See: https://reactrouter.com/en/main/routers/router-provider.

The change from v5 to v6 was pretty big, so we can always switch back to v5 if it's simpler?

Reply:
No don't switch it back to v5. People should be using the latest versions of stuff. Also we don't need to explain this sort of stuff because it's just the boilerplate. Can just blackbox all of it like `public static void main(String[] args)` in java lol